tfsdk/schema.go

```diff
@@ -183,3 +183,20 @@ func (s Schema) tfprotov6Schema(ctx context.Context) (*tfprotov6.Schema, error)
 
 	return result, nil
 }
+
+// validate performs all Attribute validation.
+func (s Schema) validate(ctx context.Context, req ValidateSchemaRequest, resp *ValidateSchemaResponse) {
+	for name, attribute := range s.Attributes {
+		attributeReq := ValidateAttributeRequest{
+			AttributePath: tftypes.NewAttributePath().WithAttributeName(name),
+			Config:        req.Config,
+		}
+		attributeResp := &ValidateAttributeResponse{
+			Diagnostics: resp.Diagnostics,
+		}
+
+		attribute.validate(ctx, attributeReq, attributeResp)
+
+		resp.Diagnostics = attributeResp.Diagnostics
```

Comment:
Should this append instead, in case multiple attribute validators return warning diags?

Reply:
Initially this was appending, but I was asked to switch it to "passthrough" diagnostics: https://github.com/hashicorp/terraform-plugin-framework/pull/75#discussion_r678710013

Personally I think pre-populating a response, while offering overwrite capabilities easily, could be confusing for implementors to understand the semantics. We could offer prior diagnostics in the request with diagnostic modifications in the response to clarify things on both sides of these calls (or not allow diagnostic modification). Please let me know if you would like me to raise an issue regarding this.